### PR TITLE
Fix remaining mutants, part 3

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #52

### CODE REVIEW

File: src/Locale.php [hunk 1]
```diff
@@ -230,13 +230,13 @@ private function isRequestIgnored(ServerRequestInterface $request): bool
     }
 
     /**
-     * @psalm-assert array<string, string> $this->supportedLocales
+     * @psalm-assert array<string, string> $supportedLocales
      *
      * @throws InvalidLocalesFormatException
      */
-    private function assertLocalesFormat(): void
+    private function validateSupportedLocales(array $supportedLocales): void
```
Comment (samdark) [inline src/Locale.php:237]: When the validation function throws an exception it is usually called `assert*()`.

File: src/Locale.php [hunk 2]
```diff
@@ -45,29 +50,30 @@ final class Locale implements MiddlewareInterface
...
         private UrlGeneratorInterface $urlGenerator,
         private SessionInterface $session,
         private LoggerInterface $logger,
         private ResponseFactoryInterface $responseFactory,
-        private array $supportedLocales = [],
+        array $supportedLocales = [],
         private array $ignoredRequestUrlPatterns = [],
         private bool $secureCookie = false,
     ) {
         $this->cookieDuration = new DateInterval('P30D');
```
[the remark this answers is not in the document]
Comment (samdark) [reply src/Locale.php:65]: Could be useful, yes.

File: src/HttpCache.php [hunk 3]
```diff
@@ -28,12 +28,12 @@ final class HttpCache implements MiddlewareInterface
     /**
      * @var callable|null
      */
-    private $lastModified = null;
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/HttpCache.php:31]: > [EA] Null assignment can be safely removed. Define null in annotations if it's important.

File: src/HttpCache.php [hunk 4]
```diff
@@ -28,12 +28,12 @@ final class HttpCache implements MiddlewareInterface
...
+    private $lastModified;
 
     /**
      * @var callable|null
      */
-    private $etagSeed = null;
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/HttpCache.php:36]: > [EA] Null assignment can be safely removed. Define null in annotations if it's important.

File: src/HttpCache.php [hunk 5]
```diff
@@ -185,16 +185,15 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
...
                 str_replace('-gzip', '', $request->getHeaderLine(Header::IF_NONE_MATCH)),
-                -1,
-                PREG_SPLIT_NO_EMPTY,
+                flags: PREG_SPLIT_NO_EMPTY,
             );
 
             // "HTTP_IF_NONE_MATCH" takes precedence over "HTTP_IF_MODIFIED_SINCE".
             // https://tools.ietf.org/html/rfc7232#section-3.3
-            return $etag !== null && !empty($header) && in_array($etag, $header, true);
+            return $etag !== null && in_array($etag, $headerParts, true);
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/HttpCache.php:196]: Need a case with this `$pattern` and when $subject is guaranteed to be a string.

File: src/HttpCache.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (vjik) [reply src/HttpCache.php:31]: Yes. It's work for non-typed properties and don't work with typed properties. Suggest always add default values for more clarify and uniformity.

File: src/HttpCache.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (vjik) [reply src/HttpCache.php:36]: waiting https://github.com/yiisoft/yii-middleware/pull/63#discussion_r1188411831

File: src/HttpCache.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (arogachev) [reply src/HttpCache.php:196]: OK, a case is needed anyway to test this properly without decreasing coverage and appearing of new mutants.

[the remark this answers is not in the document]
Comment (arogachev) [reply src/HttpCache.php]: Still no test case have been found. Added change without it not to block merging.

**Review by vjik — APPROVED** [review]

**Pull request merged**